flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManagerBase.java

```diff
@@ -131,19 +131,22 @@ public abstract class FileMergingSnapshotManagerBase implements FileMergingSnaps
     /**
      * The {@link DirectoryStreamStateHandle} for shared state directories, one for each subtask.
      */
+    @GuardedBy("lock")
```

Comment:
If this map is guarded by `lock`, the methods `registerSubtaskForSharedStates` and `unregisterSubtask` should also be protected by `lock`

Reply:
Removed the `@GuardedBy("lock")`, cause the map is `ConcurrentHashMap`, which is thread-safe.

Reply:
Well, the `fileSystemInitiated`, `checkpointDir`, `sharedStateDir ` and `taskOwnedStateDir` would also be `@GuardedBy("lock")`. I'd suggest removing this since this is controlling one-time initialization, not a typical concurrent modification scenario.

Reply:
Or adding @GuardedBy("lock") to each of those above variables would be fine with me.

Reply:
Thanks for the suggestion, removed `@GuardedBy("lock")`.